ggml-opencl.cpp

```diff
@@ -1835,7 +1835,10 @@ static void ggml_cl_mul_mat_q_f32(const ggml_tensor * src0, const ggml_tensor *
                     CL_CHECK(clEnqueueNDRangeKernel(queue, *to_fp32_cl, 1, &offset, &global, local > 0 ? &local : NULL, events.size(), !events.empty() ? events.data() : NULL, NULL));
                 }
 
-                for (int64_t i12 = i02 * r2, e12 = i12 + r2; i12 < e12; i12++) {
+                int64_t i12 = i02 * r2;
+                int64_t e12 = i12 + r2;
+                events.reserve(e12 - i12);
```

Comment:
For future reference: `events` is cleared at the end of this inner loop, so its actual maximum capacity is `3`. Even ignoring the `clear()`, `reserve()` does not grow the vector by the specified amount, it increases the capacity *to* the specified amount&mdash;so you would need to reserve `events.size() + e12 - i12` instead, if you were to even bother.

Luckily, this file is gone now, so this particular instance doesn't matter. But we should be more careful going forward.

Reply:
@cebtenzzre, good catch. More reviewers there are, lower chance making a mistake, you're right.